sources/logdog/src/log_request.rs

```diff
@@ -196,21 +198,72 @@ where
             request: request.to_string()
         }
     );
-    let dest = tempdir.as_ref().join(request.filename);
-    let _ = fs::copy(&request.instructions, &dest).with_context(|| error::FileCopy {
-        request: request.to_string(),
-        from: request.instructions,
-        to: &dest,
+    let mut glob_paths = glob(request.instructions).context(error::ParseGlobPattern {
+        pattern: request.instructions.to_string(),
+    })?;
+    let multiple_files = glob_paths.count() > 1;
+    // iterator gets consumed in count, therefore re-create
+    glob_paths = glob(request.instructions).context(error::ParseGlobPattern {
+        pattern: request.instructions.to_string(),
     })?;
+    for entry in glob_paths {
+        if let Ok(path) = entry {
+            ensure!(path.is_file(), error::Directory { path });
+            // if there are more than one file then all the files will be written to same file,
+            // therefore keep destination filename same as source filename
```

Comment:
I think trying to fit globs into the `file` mode is causing more trouble than it's worth, with the iterator count and file naming issues.  I'd suggest we switch to a new `glob /bla/*` log request format.

Either way, we should handle the possibility of multiple targets with the same name, which would currently overwrite each other, and wasn't a problem before because the target names were all written in the config file.  I think the simplest, clearest solution is probably to keep the full paths to the globbed files, i.e. /a/b/file would show up in the tarball at /a/b/file.

Reply:
My reasoning behind modifying file mode was: 
1. The glob in the end is going to target files, so it is a file based request.
2. Current request conf format is `mode filename instruction`, so i was not sure how to fit glob based requirement in this. Your suggestion of `glob /bla/*` will consider `/bla/*` pattern as filename in request argument which I thought was weird because `/bla/*` is actually an instruction of the source file. I had multiple thought of using `glob * /bla/*` all seemed weird to me. So, I finally settled on counting the number of files and making decision and ignore filename argument for multiple files.

Let me know what you think about this.

`/a/b/file` could be useful. I can do this, regardless of above conversation.

Reply:
>     2. Current request conf format is `mode filename instruction`, so i was not sure how to fit glob based requirement in this. Your suggestion of `glob /bla/*` will consider `/bla/*` pattern as filename in request argument which I thought was weird because `/bla/*` is actually an instruction of the source file. I had multiple thought of using `glob * /bla/*` all seemed weird to me. So, I finally settled on counting the number of files and making decision and ignore filename argument for multiple files.

We can make `handle_log_request` check the first key in the line before parsing the request, to make the number of fields dynamic.  They don't *have* to have three fields per line, that was just convention for existing modes.

Reply:
makes sense 👍

Reply:
I agree with @tjkirch here. I think a separate 'command' for glob is more intuitive.

Reply:
changed :)